### PR TITLE
Configure LGTM for automated code reviews

### PR DESCRIPTION
**Pull request by tdonohue**
## Description
This PR configures [LGTM](https://lgtm.com/) for our DSpace backend by adding a new `.lgtm.yml` configuration file in the root directory.  This is to fix current LGTM failures in all PRs against our `master` branch.

At this time the `.lgtm.yml` file only configures the following:
* Tell LGTM that we require Java 11 to build the project

This PR will trigger an automated build/test in LGTM so that we can determine whether it's working now.

### CODE REVIEW

**Comment by tdonohue** [comment]
This PR has **fixed the LGTM** build issues, even though the `LGTM analysis: Java` check responded with `Analysis failed (could not build the base commit (a8c3965))`.  

The reason for this partial failure is that LGTM does **two builds**:
1. First, it builds the PR and analyzes it.  That **succeeded** for this PR
2. Then, it builds current `master` (i.e. "base commit") to compare to the PR in order to see what has changed.  That step **failed** because LGTM is failing on current master.

So, long story short, this PR fixes the LGTM issues. Once merged, all PRs on `master` showing failures from LGTM just need to be rebased.
